File: octave/loop/leap_year.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Author  : Bhishan Poudel
# Date    : May 23, 2016
# Ref     : https://www.rosettacode.org/wiki/Leap_year#Python

# Function to print leap year
# function to print leap year
def print_leap_year(year,value):
    if value == True:
	print("The given year",year,"is leap year.")

    else:
	print("The given year",year,"is not a leap year.")

##=============================================================================
# method 1
import calendar
year = 2001
value = calendar.isleap(year)
print_leap_year(year,value)

##=============================================================================
# method 2
def is_leap_year(year):
    if year % 100 == 0:
        return year % 400 == 0
    return year % 4 == 0


year = 2016
value = is_leap_year(year)
print_leap_year(year,value)

##=============================================================================
# method 3
import datetime

def is_this_leap_year(year):
    try:
        datetime.date(year, 2, 29)  # second month has 29 days?
    except ValueError:
        return False
    return True

year = input("Enter the year: ")  # input for numbers, raw_input for strings
value = is_this_leap_year(year)
print_leap_year(year,value)

File: octave/loop/leap_year.m
#!/usr/bin/octave -qf
% Author    : Bhishan Poudel
% Date      : May 25, 2016


% method 1
%year = 1998;  % 1900 1996 1998 2000
year = input ("Enter the year e.g. 1900 \n"); % 1900

if( mod(year,4) == 0 )
  fprintf("\nThe year %d is divisible by 4. \nIt may be a leap year.\n",year);
  if( mod(year,100) == 0 )
    % if condition is true then print the following
    fprintf('The year %d is century year.\n', year );
    if(  mod(year,400) == 0  )
      fprintf("The year %d is century year and divisible by 400.\n",year);
      fprintf("The year %d is leap year.\n",year);

    else
      fprintf("The year %d is century year, but not divisible by 400.\n",year);
      fprintf("The year %d is not a century year.\n",year);

    end
  else
    fprintf("The year %d is divisible by 4, but not a century year.\n",year);
    fprintf("The year %d is leap year.\n", year);
  end

else
  fprintf("\nThe given year %d is not divisible by 4.", year);
  fprintf("\nThe given year %d is not a leap year.\n",year);

end

% method 2
% https://www.rosettacode.org/wiki/Leap_year#MATLAB_.2F_Octave
%MATLAB, conveniently, provides a function that returns the last day of an
%arbitrary month of the calendar given the year. Using the fact that February is
%29 days long during a leap year, we can write a one-liner that solves this task.


fprintf("\n");
year = 1996
logic = (eomday(year,2) == 29);
if (logic == 1)
  fprintf("\nThe year %d is a leap year.",year);

else
  fprintf("\nThe year %d is not a leap year.", year);

end



